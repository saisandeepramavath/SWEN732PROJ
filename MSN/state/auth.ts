import {atom} from 'recoil'

import { FirebaseAuthTypes } from '@react-native-firebase/auth';

export const authState = atom<{
  isAuth: boolean;
  user: FirebaseAuthTypes.User | null;
}>({
  key: 'authState',
  default: {
    isAuth: false,
    user: null,
  },
});

export const userState = atom({
    key: 'userState',
    default: {
        name: '',
        email: '',
        photo: '',
        uid: '',
        groups: [],
    }
})

export const groupsState = atom({
    key: 'groupsState',
    default: [],
})
export const selectedGroupState = atom({
    key: 'selectedGroupState',
    default: null,
})
export const selectedGroupMembersState = atom({
    key: 'selectedGroupMembersState',
    default: [],
})
export const selectedGroupTransactionsState = atom({
    key: 'selectedGroupTransactionsState',
    default: [],
})