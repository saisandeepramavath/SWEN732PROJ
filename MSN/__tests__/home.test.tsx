import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import Home from '../app/(auth)/home';
import auth from '@react-native-firebase/auth';

jest.mock('@react-native-firebase/auth');

describe('Home', () => {
	it('should match snapshot', () => {
		const { toJSON } = render(<Home />);
		expect(toJSON()).toMatchSnapshot();
	});

	it('should display user email', () => {
		const mockUser = { email: 'test@example.com' };
		auth().currentUser = mockUser;

		const { getByText } = render(<Home />);
		expect(getByText(`Welcome back back ${mockUser.email}`)).toBeTruthy();
	});

	it('should call signOut on button press', async () => {
		const mockSignOut = jest.fn();
		auth().signOut = mockSignOut;

		const { getByText } = render(<Home />);
		fireEvent.press(getByText('Sign out'));

		await waitFor(() => {
			expect(mockSignOut).toHaveBeenCalled();
		});
	});
});
