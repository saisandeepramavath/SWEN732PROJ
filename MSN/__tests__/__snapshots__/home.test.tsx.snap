// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<View>
  <Text>
    Welcome back back 
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Sign out
      </Text>
    </View>
  </View>
</View>
`;
